src/lib/operators/export_binary.cpp

```diff
@@ -164,6 +174,56 @@ void ExportBinary::_write_chunk(const std::shared_ptr<const Table>& table, std::
   }
 }
 
+void ExportBinary::_write_partitioning_header(const std::shared_ptr<const Table>& table, std::ofstream& ofstream) {
+  const auto partition_schema = table->get_partition_schema();
+  _export_value(ofstream, partition_schema_type_to_uint.left.at(partition_schema->get_type()));
+  _export_value(ofstream, partition_schema->partition_count());
+
+  switch (partition_schema->get_type()) {
+    case PartitionSchemaType::Hash: {
+      const auto hash_schema = std::dynamic_pointer_cast<const HashPartitionSchema>(partition_schema);
```

Comment:
No point in using a dynamic pointer cast if you don't check for `nullptr`. static works just fine here.